libfreerdp/core/tcp.c

```diff
@@ -1097,7 +901,9 @@ int freerdp_tcp_connect_multi(char** hostnames, UINT32* ports, int count, int po
 		}
 	}
 
-	waitStatus = WaitForMultipleObjects(count, events, FALSE, timeout * 1000);
+	events[count] = context->abortEvent;
+
+	waitStatus = WaitForMultipleObjects(count + 1, events, FALSE, timeout * 1000);
```

Comment:
It seems to me that the event[] is never initialized after calloc. That means all events in this array is just NULL and the previous 'WSAEventSelect' doesn't take any effect(it just returns SOCKET_ERROR).

This is not introduced by this PR. It's the original logic in freerdp_tcp_connect_multi for win32.